agent/engine/docker_container_engine.go

```diff
@@ -365,6 +358,27 @@ func (dg *dockerGoClient) pullImage(image string, authData *api.RegistryAuthenti
 	return nil
 }
 
+// ImportLocalEmptyVolumeImage imports a locally-generated empty-volume image for supported platforms.
+func (dg *dockerGoClient) ImportLocalEmptyVolumeImage() DockerContainerMetadata {
+	timeout := dg.time().After(pullImageTimeout)
+
+	response := make(chan DockerContainerMetadata, 1)
+	go func() {
+		err := dg.createScratchImageIfNotExists()
+		var wrapped engineError
+		if err != nil {
+			wrapped = CreateEmptyVolumeError{err}
+		}
+		response <- DockerContainerMetadata{Error: wrapped}
+	}()
+	select {
+	case resp := <-response:
+		return resp
+	case <-timeout:
+		return DockerContainerMetadata{Error: &DockerTimeoutError{pullImageTimeout, "pulled"}}
```

Comment:
Can the `transition` string be a const ?

Reply:
Any particular reason why?  I generally like constants if there's string reuse, string comparison, or if it aids clarity.  I suppose reuse might apply here, but it's only used in two places right now.

Reply:
I was primarily concerned with reuse but it was more of a suggestion.